content/docs/hooks-intro.md

```diff
@@ -25,86 +25,86 @@ function Example() {
+Często musieliśmy utrzymywać komponenty, które z początku proste, urosły do rangi niemożliwego do utrzymania bałaganu logiki związanej ze stanem i efektów ubocznych (ang. *side effects*). Każda metoda cyklu życia zawiera zwykle mieszankę niepowiązanej ze sobą logiki. Na przykład, komponenty mogą pobierać dane w `componentDidMount` i `componentDidUpdate`. Jednakże metoda `componentDidMount` może też zawierać logikę, która tworzy słuchaczy zdarzeń. Następnie są oni czyszczeni w `componentWillUnmount`. Wzajemnie powiązany kod zostaje podzielony pomiędzy różne metody, a z kolei zupełnie niezwiązany ze sobą kod trafia do jednej. Otwiera to zbyt wielkie pole do błędów i niekonsekwencji.
 
-In many cases it's not possible to break these components into smaller ones because the stateful logic is all over the place. It's also difficult to test them. This is one of the reasons many people prefer to combine React with a separate state management library. However, that often introduces too much abstraction, requires you to jump between different files, and makes reusing components more difficult.
+Wielokrotnie zdarza się, że nie ma możliwości rozbicia tych komponentów na mniejsze części, ponieważ logika związana ze stanem jest już wszędzie. Trudno jest też je testować. Jest to jeden z powodów, dla których wielu woli połączyć Reacta z zewnętrzną biblioteką do zarządzania stanem. To jednak często wprowadza zbyt wiele abstrakcji, zmusza do skakania pomiędzy plikami i utrudnia ponowne wykorzystanie komponentów.
 
-To solve this, **Hooks let you split one component into smaller functions based on what pieces are related (such as setting up a subscription or fetching data)**, rather than forcing a split based on lifecycle methods. You may also opt into managing the component's local state with a reducer to make it more predictable.
+Aby rozwiązać ten problem, **Hooki pozwalają podzielić komponent na mniejsze funkcje, bazując na powiązanych ze sobą częściach (takich jak tworzenie subskrypcji, czy pobieranie danych)**, zamiast wymuszać sztuczny podział, związany z metodami cyklu życia. Ewentualnie, aby uczynić to bardziej przewidywalnym, możesz też dzięki nim oddelegować zarządzanie lokalnym stanem komponentu do reduktora.
 
-We'll discuss this more in [Using the Effect Hook](/docs/hooks-effect.html#tip-use-multiple-effects-to-separate-concerns).
+Szerzej omówimy to w rozdziale [Używanie Hooka Efektu](/docs/hooks-effect.html#tip-use-multiple-effects-to-separate-concerns).
```

Comment:
Hook Efektu? Hmm, trochę mi to nie leży. Nie mam innego pomysłu na to, ale chętnie poczekam na zdanie innych osób.

Reply:
Muszę się z tym przespać ;)

Reply:
Jeżeli side effects przetłumaczymy na efekty uboczne, to ten hook pozwala na wykonywanie efektów ubocznych. Więc jest Hookiem Efektu :).

Reply:
Możemy też iść w stronę "skutku ubocznego"
https://pl.wikipedia.org/wiki/Skutek_uboczny_(informatyka)

Reply:
Wiem, że w oryginale tego tam nie ma, ale czy możemy dodać ten drugi człon? "Hook Skutku Ubocznego"? Samo "Skutku" nic nie mówi...

Reply:
Samo _Effect_ też nie wiele mówi :). Nie chciałbym poprawiać twórców oryginału.

Reply:
Tłumacząc kolejną sekcję dochodzę do wniosku, że "efekt uboczny" będzie lepszy. Budzi on od razu skojarzenie z nazwą hooka - `useEffect`.